src/axom/spin/BVH.hpp

```diff
@@ -165,13 +170,21 @@ class BVH
    * \pre boxes != nullptr
    * \pre numItems > 0
    */
-  BVH( const FloatType* boxes, IndexType numItems );
+  BVH( const FloatType* boxes, IndexType numItems,
```

Comment:
OK. So I see this here. Is there a reason we don't want to just have this just expose the Umpire allocator stuff. Why are we wrapping it?

Reply:
We are not wrapping it, we just expose the integer ID of the umpire allocator instead of exposing umpire data-structures in the API. My preference is not to leak umpire data-structures in the API. 

- Is there a good reason that we should? 

- Are there any limitations with the way it is done currently?

To the best of my knowledge, most host-codes use the integer ID to specify an allocator.

Although this class strictly requires `RAJA` and `Umpire`, Axom can be compiled without `RAJA` and `Umpire`. So, limiting the interface to just basic primitive types make the API cleaner and avoids the otherwise needed `ifdef`.

Reply:
Got it. It was unclear to me what was going on until I looked at this part of the code. I think the way you have done it makes sense. What happens if I build Axom without RAJA and Umpire and try to use this class. Is it unavailable and the application code will not compile?

Reply:
Correct -- This class is only included if `Umpire` and `RAJA` are available. So, if the host code is using this class, it won't compile if the Axom is not built with `RAJA` and `Umpire`. If they are pointing to an axom installation they would get an error be able to find the include.

In retrospect, we do have the following in the header:
https://github.com/LLNL/axom/blob/fb2bb2a1f4e37e8a965e8799bca7896b64084e3a/src/axom/spin/BVH.hpp#L19-L21

So, we may want to install the header regardless so that they get a more meaningful error message when they try to use the class and haven't compiled Axom correctly. Thoughts?

We can also make it work without `RAJA` and `Umpire` as well. To do so, it would require us having our own abstraction layer to do `scans` and `reductions` so that the code compiles without `RAJA`.

Reply:
See #238 .